src/Bundle.js

```diff
@@ -87,89 +88,85 @@ export default class Bundle {
 			});
 	}
 
+	// TODO would be better to deconflict once, rather than per-render
 	deconflict ( es6 ) {
-		let definers = blank();
-		let conflicts = blank();
+		let usedNames = blank();
+
+		// ensure no conflicts with globals
+		keys( this.assumedGlobals ).forEach( name => usedNames[ name ] = true );
+
+		let allReplacements = blank();
 
 		// Assign names to external modules
 		this.externalModules.forEach( module => {
-			let name = makeLegalIdentifier( module.suggestedNames['*'] || module.suggestedNames.default || module.id );
-
-			while ( definers[ name ] ) {
-				conflicts[ name ] = true;
-				name = `_${name}`;
-			}
+			// while we're here...
+			allReplacements[ module.id ] = blank();
 
-			definers[ name ] = [ module ];
-			module.name = name;
-			this.assumedGlobals[ name ] = true;
+			// TODO is this necessary in the ES6 case?
+			let name = makeLegalIdentifier( module.suggestedNames['*'] || module.suggestedNames.default || module.id );
+			module.name = getSafeName( name );
```

Comment:
It isn't necessary in the ES6 case, but if we always do it, taking the needs of all module formats into account, we only have to deconflict once. Might as well do it in `Bundle.build`.